lib/presentation/manager/quota/verify_quota_manager.dart

```diff
@@ -36,18 +36,36 @@
 
 import 'package:dartz/dartz.dart';
 import 'package:domain/domain.dart';
+import 'package:linshare_flutter_app/presentation/redux/actions/account_action.dart';
 import 'package:linshare_flutter_app/presentation/redux/actions/upload_file_action.dart';
 import 'package:linshare_flutter_app/presentation/redux/states/app_state.dart';
 import 'package:redux/redux.dart';
 
 class VerifyQuotaManager {
   final Store<AppState> _store;
   final GetQuotaInteractor _getQuotaInteractor;
+  final GetAuthorizedInteractor _getAuthorizedInteractor;
 
-  VerifyQuotaManager(this._store, this._getQuotaInteractor);
+  VerifyQuotaManager(this._store, this._getQuotaInteractor, this._getAuthorizedInteractor);
 
   Future<bool> hasEnoughQuotaAndMaxFileSize({List<FileInfo> filesInfos}) async {
-    final quotaUuid = _store.state.account.user.quotaUuid;
+    var quotaUuid = QuotaId('');
+
+    if (_store.state.account.user != null) {
+      quotaUuid = _store.state.account.user.quotaUuid;
+    } else {
+      await _getAuthorizedInteractor.execute()
+          .then((result) => result.fold(
+            (left) => null,
+            (right) {
+              if (right is GetAuthorizedUserViewState) {
+                quotaUuid = right.user.quotaUuid;
+                _store.dispatch(SetAccountInformationsAction(right.user));
+              }
+            }
+          ));
```

Comment:
why not ignore it, and make the upload continue?

Reply:
I think it is natural to wait. We have a loading screen to let the user know that the upload is being made. And this process is very fast. If there is a failure or an error, it will also break for the user to continue. More if we do this. Then we can continue to correctly get the quota, and the user information is updated for later use.

Reply:
> why not ignore it, and make the upload continue?

I think your way is too a dodge solution. When I know for sure that the user has failed.